contents/security/passwordStorage.md

```diff
@@ -0,0 +1,118 @@
+
+### What's wrong with <trigger for="pop:plaintext">plaintext</trigger>?
+
+<popover id="pop:plaintext" title="_Plaintext_ refers to unencrypted information" placement="top">
+  <div slot="content">
+It is a cryptography term generally referring to text before encyption or after decrypting it. Another term for it is cleartext.
+  </div>
+</popover>
+
+Since usernames are typically public information, it is OK to store them in plaintext. However, passwords are typically secret information and hence, cannot be stored in plaintext.
+
+If passwords are stored in plaintext:
+* Should your database be compromised, attackers can easily use the stored plaintext passwords to login as the user and potentially access other functionality. This grants the attacker more power than just reading values off the database. If the passwords aren't even in plaintext, the attacker might not be able to read the passwords.
+* Your employees can see the plaintext passwords of your users. If your employees are ever compromised, they might be willing to simply leak the passwords.
+* It is common for users to re-use the same password on multiple sites. If your passwords are stored as plaintext and accessed, attackers can easily re-use the same passwords on other websites with sensitive data (such as banking websites) and gain access to those.
+
+To prevent these situations from occuring, it is good to perform hashing on passwords. Unlike <trigger for="pop:encrypt">encryption</trigger>, hashing is a <trigger for="pop:oneway">one-way function</trigger>. This means that an attacker cannot decrypt the encrypted password to retrieve the original password, thus ensuring that none of the aforementioned vulnerabilities are possible.
+
+<popover id="pop:encrypt" title="Encryption refers to the process of changing a message such that it becomes essentially random text." placement="top">
+  <div slot="content">
+ The opposite process is called decryption, which converts the text back into the original message.
+  </div>
+</popover>
+
+<popover id="pop:oneway" title="" placement="top">
+<div slot="content">
+	A one way function is a function that is easy to compute in one way, but is close to impossible to compute the other way.
+</div>
+</popover>
+
+## What is hashing?
+
+Hashing is the act of transforming a set of data into another set of data that is impossible to reverse. As an example, suppose you wanted to hash [The Complete Works of Shakespeare](https://www.gutenberg.org/files/100/100-0.txt). If you performed a SHA1 hash on the entire contents of that file, including the licensing information at the very end, you'll end up hashing the entire 157257 lines of that file into `C9B38187EA0CC82B72142B92034DA859A3550B57`. Obviously, it is impossible to condense all 157257 lines of text into just those 40 characters. 
```

Comment:
Also: only this text file will hash to `C9B38187EA0CC82B72142B92034DA859A3550B57`. Another text file, or even changing one character in this file, will result in a dramatically different hash. Otherwise, the reader might think that other files (or passwords!) could also produce this value.